### PR TITLE
Resize LVM VG

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/dsn6CQmd/204-3-lvm-exper-partitioner-resize-a-vg

### CODE REVIEW

File: src/lib/y2partitioner/actions/controllers/lvm_vg.rb [hunk 1]
```diff
@@ -53,12 +53,33 @@ class LvmVg
...
+        # @note The returned title depends on the action to perform (see {#initialize_action})
+        #
+        # @return [String]
+        def wizard_title
+          case action
+          when :add
+            # TRANSLATORS: dialog title when creating a logical volume
+            _("Add Volume Group")
+          when :resize
+            # TRANSLATORS: dialog title when resizing a logical volume, where %s is replaced
```
Comment (ancorgs) [inline src/lib/y2partitioner/actions/controllers/lvm_vg.rb]: When resizing **an LVM volume group**. Also the example looks like a RAID name.

File: src/lib/y2partitioner/actions/controllers/lvm_vg.rb [hunk 2]
```diff
@@ -77,6 +98,13 @@ def vg_size
           vg.size
         end
 
+        # Size of the logical volumes belonging to the current volume group
+        #
+        # @return [Y2Storage::DiskSize]
+        def lvs_size
+          vg.lvm_lvs.map(&:size).inject(Y2Storage::DiskSize.zero, :+)
```
Comment (ancorgs) [inline src/lib/y2partitioner/actions/controllers/lvm_vg.rb]: NP: just for your info, there is a `DiskSize.sum` method. In general it's slightly nicer, like
```ruby
Y2Storage::DiskSize.sum(vg.lvm_lvs.map(&:size))
```

Not a big deal, anyways.

File: src/lib/y2partitioner/actions/resize_lvm_vg.rb [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+# find current contact information at www.suse.com.
+
+require "yast"
+require "y2partitioner/actions/transaction_wizard"
+require "y2partitioner/actions/controllers/lvm_vg"
+require "y2partitioner/dialogs/lvm_vg_resize"
+
+module Y2Partitioner
+  module Actions
+    # Action for resizing a new LVM volume group
```
Comment (ancorgs) [inline src/lib/y2partitioner/actions/resize_lvm_vg.rb]: Resizing a **new** volume group? Isn't that kind of a contradiction?

File: src/lib/y2partitioner/widgets/lvm_vg_devices_selector.rb [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+      # Validates that unselected phyical volumes can be removed
+      #
+      # @note A physical volume cannot be removed if it is already committed. An error
+      #   popup is shown when a committed physical volume is tried to be removed.
+      #
+      # @see Actions::Controllers::LvmVg#committed_devices
+      #
+      # @param sids [Array<Integer>]
+      # @return [Boolean]
+      def validate_unselect(sids)
```
Comment (ancorgs) [inline src/lib/y2partitioner/widgets/lvm_vg_devices_selector.rb]: NP: not sure whether it makes sense to call this method `validate_xxx` and make it return a boolean. It's only real usage is displaying messages to the user but is not used to interrupt the option or something like that (in fact, the return value is ignored).

I can live with it, but it's kind of confusing.

File: test/y2partitioner/widgets/device_resize_button_test.rb [hunk 5]
```diff
@@ -170,6 +170,40 @@
...
+            .and_return(action_result)
+        end
+
+        let(:device) { Y2Storage::LvmVg.find_by_vg_name(current_graph, "vg0") }
+
+        let(:action_result) { nil }
+
+        let(:scenario) { "lvm-two-vgs.yml" }
+
+        it "performs the action for resizing a partition" do
```
Comment (ancorgs) [inline test/y2partitioner/widgets/device_resize_button_test.rb]: Resizing a partition?

File: src/lib/y2partitioner/actions/controllers/lvm_vg.rb (hunk 2 above)
Comment (joseivanlopez) [reply src/lib/y2partitioner/actions/controllers/lvm_vg.rb]: Nice to know :)

File: src/lib/y2partitioner/widgets/lvm_vg_devices_selector.rb (hunk 4 above)
Comment (joseivanlopez) [reply src/lib/y2partitioner/widgets/lvm_vg_devices_selector.rb]: Yes, in this case it is only a check, not a validation.

**Pull request merged**